account_operating_unit/models/account_move.py

```diff
@@ -65,6 +65,63 @@ def _check_move_operating_unit(self):
                     )
                 )
 
+    def _check_ou_balance(self, lines):
+        # Look for the balance of each OU
+        ou_balance = {}
+        for line in lines:
+            if line.operating_unit_id.id not in ou_balance:
+                ou_balance[line.operating_unit_id.id] = 0.0
+            ou_balance[line.operating_unit_id.id] += line.credit - line.debit
+        return ou_balance
+
+    def reconcile(self):
+        # if one OU pays the invoices of different OU
+        # a regularization entry must be created (this
+        # was a feature in version <= 12)
+        if self and not self[0].company_id.ou_is_self_balanced:
+            return super().reconcile()
+        bank_journal = self.mapped("move_id.journal_id").filtered(
+            lambda l: l.type in ("bank", "cash")
+        )
+        if not bank_journal:
+            return super().reconcile()
+        bank_journal = bank_journal[0]
+        # If all move lines point to the same operating unit, there's no
+        # need to create a balancing move line
+        if len(self.mapped("operating_unit_id")) <= 1:
+            return super().reconcile()
+        # Create balancing entries for un-balanced OU's.
+        move_vals = self._prepare_inter_ou_balancing_move(bank_journal)
+        move = self.env["account.move"].create(move_vals)
+        ou_balances = self._check_ou_balance(self)
```

Comment:
```suggestion
        ou_balances = self.move_id._check_ou_balance(self.move_id)
```

Reply:
and delete above _check_ou_balance

Reply:
This is not the same code. In the reconcile method I am calling that method with 2 separate account.move.line records from different account moves.